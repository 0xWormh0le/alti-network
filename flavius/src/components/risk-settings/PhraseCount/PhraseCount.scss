@import 'styles/_shared.scss';

.PhraseCount {
  display: flex;
  justify-content: flex-end;
  color: $LIGHTER_GREY;

  &__loading {
    width: 15px;
    height: 11px;
  }
}
