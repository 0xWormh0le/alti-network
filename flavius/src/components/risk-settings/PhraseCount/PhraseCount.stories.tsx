import React from 'react'
import { Story, Meta } from '@storybook/react/types-6-0'
import PhraseCount, { PhraseCountProps } from './PhraseCount'

export default {
  title: 'RiskSettings/PhraseCount',
  component: PhraseCount
} as Meta

const Template: Story<PhraseCountProps> = (args) => <PhraseCount {...args} />

export const Default = Template.bind({})
Default.args = {
  count: 4
}

export const Loading = Template.bind({})
Loading.args = {
  loading: true
}
