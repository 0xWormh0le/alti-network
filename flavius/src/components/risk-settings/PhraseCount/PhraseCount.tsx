import React from 'react'
import ContentLoader from 'react-content-loader'

import Typography, { TypographyVariant } from 'components/elements/Typography'
import Tooltip from 'components/widgets/Tooltip'
import UI_STRINGS from 'util/ui-strings'
import './PhraseCount.scss'

export const TOTAL_PHRASE_COUNT = 20

export interface PhraseCountProps {
  count: number
  loading?: boolean
}

const PhraseCountLoading = () => (
  <ContentLoader
    backgroundColor='#eeeeee'
    foregroundColor='#fdfdfd'
    preserveAspectRatio='none'
    height={11}
    width={15}
    uniqueKey='phrasecount-loading'
    className='PhraseCount__loading'>
    <rect x={0} y={0} width={15} height={11} rx={2} ry={2} />
  </ContentLoader>
)

export const PhraseCount: React.FC<PhraseCountProps> = ({ loading, count }) => {
  return (
    <Typography variant={TypographyVariant.BODY_SMALL} className='PhraseCount'>
      <Tooltip text={UI_STRINGS.RISK_SETTINGS.MAXIMUM_PHASES}>
        <span>
          {loading ? <PhraseCountLoading /> : count} of {TOTAL_PHRASE_COUNT} available phrases
        </span>
      </Tooltip>
    </Typography>
  )
}

export default PhraseCount
