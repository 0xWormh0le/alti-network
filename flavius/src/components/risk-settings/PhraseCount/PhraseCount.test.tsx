import React from 'react'
import { render } from '@testing-library/react'
import { PhraseCount, PhraseCountProps } from './PhraseCount'

describe('PhraseCount', () => {
  it('renders correctly', () => {
    const props: PhraseCountProps = {
      count: 10,
      loading: false
    }
    const { container } = render(<PhraseCount {...props} />)
    expect(container).toMatchSnapshot()
  })

  it('renders loading status correctly', () => {
    const props: PhraseCountProps = {
      count: 10,
      loading: true
    }
    const { container } = render(<PhraseCount {...props} />)
    expect(container).toMatchSnapshot()
  })
})
