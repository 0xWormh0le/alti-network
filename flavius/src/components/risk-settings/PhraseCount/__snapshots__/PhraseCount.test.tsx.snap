// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PhraseCount renders correctly 1`] = `
<div>
  <p
    class="Typography Typography--body-small PhraseCount"
  >
    <span
      class="trigger"
    >
      10
       of 
      20
       available phrases
    </span>
  </p>
</div>
`;

exports[`PhraseCount renders loading status correctly 1`] = `
<div>
  <p
    class="Typography Typography--body-small PhraseCount"
  >
    <span
      class="trigger"
    >
      <svg
        aria-labelledby="phrasecount-loading-aria"
        class="PhraseCount__loading"
        height="11"
        preserveAspectRatio="none"
        role="img"
        width="15"
      >
        <title
          id="phrasecount-loading-aria"
        >
          Loading...
        </title>
        <rect
          clip-path="url(#phrasecount-loading-diff)"
          height="100%"
          role="presentation"
          style="fill: url(#phrasecount-loading-animated-diff);"
          width="100%"
          x="0"
          y="0"
        />
        <defs
          role="presentation"
        >
          <clippath
            id="phrasecount-loading-diff"
          >
            <rect
              height="11"
              rx="2"
              ry="2"
              width="15"
              x="0"
              y="0"
            />
          </clippath>
          <lineargradient
            id="phrasecount-loading-animated-diff"
          >
            <stop
              offset="0%"
              stop-color="#eeeeee"
              stop-opacity="1"
            >
              <animate
                attributeName="offset"
                dur="1.2s"
                keyTimes="0; 0.25; 1"
                repeatCount="indefinite"
                values="-2; -2; 1"
              />
            </stop>
            <stop
              offset="50%"
              stop-color="#fdfdfd"
              stop-opacity="1"
            >
              <animate
                attributeName="offset"
                dur="1.2s"
                keyTimes="0; 0.25; 1"
                repeatCount="indefinite"
                values="-1; -1; 2"
              />
            </stop>
            <stop
              offset="100%"
              stop-color="#eeeeee"
              stop-opacity="1"
            >
              <animate
                attributeName="offset"
                dur="1.2s"
                keyTimes="0; 0.25; 1"
                repeatCount="indefinite"
                values="0; 0; 3"
              />
            </stop>
          </lineargradient>
        </defs>
      </svg>
       of 
      20
       available phrases
    </span>
  </p>
</div>
`;
